cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
+  - `script-src-attr` controls the execution of event handlers.
+- `style-src` controls from where styles get applied to a document. This includes `<link>` elements, `@import` rules, and requests originating from a `Link` HTTP response header field.
+  - `style-src-elem` controls styles except for inline attributes.
+  - `style-src-attr` controls styles attributes.
 
-### New in CSP2
+## Document Directives
 
--   **form-action** - retricts which URLs can be used as the action of HTML form elements
--   **frame-ancestors** - indicates whether the user agent should allow embedding the resource using a frame, iframe, object, embed or applet element, or equivalent functionality in non-HTML resources
--   **plugin-types** - restricts the set of plugins that can be invoked by the protected resource by limiting the types of resources that can be embedded
--   **base-uri** - restricts the URLs that can be used to specify the document base URL
--   **child-src** (d) - governs the creation of nested browsing contexts as well as Worker execution contexts
+Document directives instruct the browser about the properties of the document to which the policies will apply to. 
 
-CSP Sample Policies
-===================
+- `base-uri` specifies the possible URLs that the `<base>` element can use.
+- `plugin-types` limits the types of resources that can be loaded into the document (*e.g.* application/pdf). 3 rules apply to the affected elements, `<embed>` and `<object>`:
+  - The element needs to explicitly declare its type.
+  - The element's type needs to match the declared type.
+  - The element's resource need to meatch the declared type.
+- `sandbox` restricts a page's actions such as submitting forms.
+  - Only applies when used with the request header `Content-Security-Policy`.
+  - Not specifying a value for the directive activates all of the sandbox restrictions. `Content-Security-Policy: sandbox;`
+  - [Sandbox syntax](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox#Syntax)
 
-Basic CSP Policy
-----------------
+## Navigation Directives
 
-This policy will only allow resources from the originating domain for all the default level directives, and will not allow inline scripts/styles to execute. If your application and function with these restrictions, it drastically reduces your attack surface having this policy in place, and will work with most modern browsers.
+Navigation directives instruct the browser about the locations that the document can navigate to.
 
-The most basic policy assumes:
+- `navigate-to` restricts the URLs which a document can navigate to by any mean.
+- `form-action` restricts the URLs which the forms can submit to.
+- `frame-ancestors` restricts the URLs that can embed the requested resource inside of  `<frame>`, `<iframe>`, `<object>`, `<embed>`, or `<applet>` elements.
+  - If this directive is specified in a `<meta>` tag, the directive is ignored.
```

Comment:
This point is not mentioned [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors), from where this information come from?

Reply:
It actually is. Read the last line in the blue paragraph of the introduction of the directive.

Reply:
OK sorry.